source/shared/core_stmt.cpp

```diff
@@ -415,7 +423,13 @@ void core_sqlsrv_bind_param( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT param_
     if( direction == SQL_PARAM_OUTPUT ) {
         switch( php_out_type ) {
             case SQLSRV_PHPTYPE_INT:
-                convert_to_long( param_z );
+                if( zval_was_long ){
+                    convert_to_string( param_z );
+                    encoding = SQLSRV_ENCODING_SYSTEM;
+                }
```

Comment:
Do any encoding issues relating to use of commas as separators crop up? I should think that convert_to_string just takes the int and converts to a string of digits, eg. 1,000,000 becomes '1000000'. But if not, do we run into any issues handling commas when calling convert_to_double or convert_to_long below?

Reply:
I've tried testing with both encodings (utf-8 and char) and they both are fine. From wheat I researched, user can only retrieve a comma separated integer by explicitly casting an int to a string. If that is the case, the php_out_type would not be INT anymore, instead it would be a string, and the problem with bigint to int conversion would not exist.

As for the second question "do we run into any issues handling commas when calling convert_to_double or convert_to_long below", this will not be a problem since the string retrieved will not have any comma separators.